packages/collection/src/collection.ts

```diff
@@ -662,6 +662,21 @@ export class Collection<K, V> extends Map<K, V> {
 		return coll;
 	}
 
+	/**
+	 * The subtract method returns a new structure containing items where the keys and values of the original structure are not present in the other.
+	 *
+	 * @param other - The other Collection to filter against
+	 */
+	public subtract<T>(other: ReadonlyCollection<K, T>): Collection<K, V> {
+		const coll = new this.constructor[Symbol.species]<K, V>();
+		for (const [k, v] of this) {
+			if (!other.has(k) && !Object.is(v, other.get(k))) {
```

Comment:
Hi, sorry for late review, I have checked this PR quite a few times, but I still can't wrap my head on this condition. Does it even work?

Let's debug this using our brains:

- `other.has(k)` returns `true`
  - It's negated, so `false && expr`, which always returns `false`.
- `other.has(k)` returns `false`
  - It's negated, so `true && expr`, which always returns `expr`.
  - `**other.get(k)` returns `V | undefined`, but because `other.has(k)` is `false`, then `other.get(k)` must always return `undefined`.**
  - That boils down to the constant `Object.is(v, undefined)`.
  - If `v` is always defined:
    - Then `Object.is(v, undefined)` is always `false`, and with the negation, it always returns `true`.
    - **And therefore, for a non-undefined `V`, the condition for substract is for `other` to not have the key.**
  - If `v` can be undefined:
    - **Then `Object.is(v, undefined)` may return `true`, but only if `v` is `undefined`.

I have written the logical flaws in bold for visibility.

---

I believe the right code would be `!other.has(k) || !Object.is(v, other.get(k))`, as `other.get` will be executed only if `other.has(k)` returns `true`.

So yeah, @vladfrangu's comment is right.